apps/spruce/src/pages/Version.tsx

```diff
@@ -44,6 +44,12 @@ export const VersionPage: React.FC = () => {
   const [redirectURL, setRedirectURL] = useState(undefined);
   const [isLoadingData, setIsLoadingData] = useState(true);
 
+  // Reset state when version ID changes.
+  useEffect(() => {
+    setIsLoadingData(true);
+    setRedirectURL(undefined);
+  }, [versionId]);
+
```

Comment:
I think adding keyArgs for the HasVersion query [here](https://github.com/evergreen-ci/ui/blob/92881dfbd8407451d20c7aaa589ac7f31d14323b/apps/spruce/src/gql/client/cache.ts#L6) also fixes the issue. IMO that's a little more elegant since it better leverages the Apollo cache!

Reply:
I found that the cache setting fixes the error but it's not completely clear to me how because the setting change represents the default behavior. Apollo Client docs mention [here](https://www.apollographql.com/docs/react/caching/cache-field-behavior/#example-1) that all of a field's arguments are keyArguments by default. I also tested hasVersion query with a network-only fetch-policy and still got the error.

Reply:
> Apollo Client docs mention [here](https://www.apollographql.com/docs/react/caching/cache-field-behavior/#example-1) that all of a field's arguments are keyArguments

This is a great point and has me confused! Glad it works though...